air-script/tests/helpers.rs

```diff
@@ -48,3 +47,29 @@ impl Test {
         Ok(codegen.generate())
     }
 }
+
+/// Parses the provided source and returns the AST.
+fn parse(source: &str) -> Result<parser::ast::Source, parser::ParseError> {
+    use miden_diagnostics::{term::termcolor::ColorChoice, *};
+    use parser::{ast, ParseError, Parser};
+    use std::sync::Arc;
+
+    let codemap = Arc::new(CodeMap::new());
+    let emitter = Arc::new(DefaultEmitter::new(ColorChoice::Auto));
+    let config = DiagnosticsConfig {
+        verbosity: Verbosity::Warning,
+        warnings_as_errors: true,
+        no_warn: false,
+        display: Default::default(),
+    };
+    let diagnostics = DiagnosticsHandler::new(config, codemap.clone(), emitter);
+    let parser = Parser::new((), codemap);
+    match parser.parse_string::<ast::Source, _, _>(&diagnostics, source) {
+        Ok(ast) => Ok(ast),
+        Err(ParseError::Lexer(err)) => {
+            diagnostics.emit(err);
+            Err(ParseError::Failed)
+        }
+        Err(err) => Err(err),
+    }
+}
```

Comment:
For sure, I just wanted to avoid changing the signature of it in this PR, because you'd generally want to be able to provide the `CodeMap` and `DiagnosticsHandler` implementations in the caller. For example, instantiate them at the start of a program and pass them in where needed, rather than instantiating them at every callsite like is done here, though there isn't anything wrong with that per se.

In my follow-on changes based on this PR, I did add some additional parse helpers that build on each other to provide a bit more flexibility to the caller without duplicating the code everywhere, so we can start using those instead, I can do a small PR that updates these redundant implementations to use those instead after that work gets merged.